src/opm/parser/eclipse/EclipseState/InitConfig/InitConfig.cpp

```diff
@@ -38,8 +39,14 @@ namespace Opm {
         return Equil( deck.getKeyword<ParserKeywords::EQUIL>(  ) );
     }
 
+    static inline FoamConfig foamconfigs( const Deck& deck ) {
```

Comment:
Instead of this logic I would let the `FoamConfig` constructor take a `Deck` - possibly as an overload.